compiler/rustc_type_ir/src/structural_impls.rs

```diff
@@ -154,19 +141,10 @@ impl<I: Interner, T: TypeVisitable<I>> TypeVisitable<I> for Vec<T> {
     }
 }
 
-impl<I: Interner, T: TypeVisitable<I>> TypeVisitable<I> for &[T] {
-    fn visit_with<V: TypeVisitor<I>>(&self, visitor: &mut V) -> ControlFlow<V::BreakTy> {
-        self.iter().try_for_each(|t| t.visit_with(visitor))
-    }
-}
-
-impl<I: Interner, T: TypeFoldable<I>> TypeFoldable<I> for Box<[T]> {
-    fn try_fold_with<F: FallibleTypeFolder<I>>(self, folder: &mut F) -> Result<Self, F::Error> {
-        self.try_map_id(|t| t.try_fold_with(folder))
-    }
-}
+// Not implemented because there's no good way to produce a new `&[T]`.
+// impl<I: Interner, T: TypeFoldable<I>> TypeFoldable<I> for &[T] { ... }
```

Comment:
It doesn't work because we actually have two impls:
```
error[E0751]: found both positive and negative implementation of trait `rustc_type_ir::fold::TypeFoldable<context::TyCtxt<'_>>` for type `&[InlineAsmTemplatePiece]`:
  --> compiler/rustc_middle/src/mir/type_foldable.rs:31:1
   |
31 | impl<'tcx> TypeFoldable<TyCtxt<'tcx>> for &'tcx [InlineAsmTemplatePiece] {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ positive implementation here
   |
   = note: negative implementation in crate `rustc_type_ir`

error[E0751]: found both positive and negative implementation of trait `rustc_type_ir::fold::TypeFoldable<context::TyCtxt<'_>>` for type `&[rustc_span::Span]`:
  --> compiler/rustc_middle/src/mir/type_foldable.rs:40:1
   |
40 | impl<'tcx> TypeFoldable<TyCtxt<'tcx>> for &'tcx [Span] {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ positive implementation here
   |
   = note: negative implementation in crate `rustc_type_ir`
```
It was these two impls that triggered me to write this commit, because I wondered why there weren't corresponding `TypeVisitable` impls.